File: Reverse Words in a String/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int left = 0, right = s.length() - 1;
        while (s[left] == ' ') {
            left++;
        }
        while (s[right] == ' ') {
            right--;
        }
        string res = "", word = "";
        for (int i = left; i <= right; i++) {
            if (s[i] == ' ') {
                while (s[i+1] == ' ') {
                    i++;
                }
                if (res == "")
                    res = word;
                else
                    res = word + " " + res;
                word = "";
            } else {
                word += s[i];
            }
        }
        if(res=="")
        return word;

        return word+" "+res;
    }
};

File: Reverse Words in a String/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        int left = 0, right = s.length() - 1;

        while (s.charAt(left) == ' ') {
            left++;
        }

        while (s.charAt(right) == ' ') {
            right--;
        }

        String res = "", word = "";

        for (int i = left; i <= right; i++) {
            if (s.charAt(i) == ' ') {
                
                while (s.charAt(i + 1) == ' ') {
                    i++;
                }
                if (res.equals("")) {
                    res = word;
                } else {
                    res = word + " " + res;
                }
                word = "";
            } else {
                word += s.charAt(i);
            }
        }

        if (res.equals("")) {
            return word;
        }

        return word + " " + res;
    }
}
